util/reportinfo.go

```diff
@@ -0,0 +1,70 @@
+/*
+This file is for detecting the user's system for helpdesk purposes.
+This is only a prototype.
+
+We will collect CPU flags and check for AVX, distro, kernel, flatpak, and config.
+More features may be added at a later time...
+
+Wael, if you would like to combine the file reading into a single function, that would be cool.
+ - lunarlattice 10/16/23
+*/
+
+package util
+
+import (
+	"errors"
+	"os"
+	"regexp"
+)
+
+type SysInfo struct {
+	AVXAvailable bool
+	Distro       string //Done
+	Kernel       string // Done
+	InFlatpak    bool   // Done
+	Config       string // Done
+}
+
+func GenerateInfo(currentConfigurationPath string) (SysInfo, error) {
+	//TODO, returns struct Sysinfo.
+	currentSystem := &SysInfo{}
+
+	// Check for AVX
+	if cpufile, err := os.ReadFile("/proc/cpuinfo"); err != nil {
+		return SysInfo{}, err
+	} else {
+		exp := regexp.MustCompile(`avx`)
```

Comment:
Parse all available CPU flags in string array, and checking for if some specific CPU flags exist, such as avx, avx2, etc.